app/routes/admin.users.$userId.tsx

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node'
+import { useFetcher, useLoaderData } from '@remix-run/react'
+import { Button, Checkbox } from '@trussworks/react-uswds'
+import { useState } from 'react'
+
+import { getUser } from './_auth/user.server'
+import {
+  addUserToGroup,
+  allGroupNames,
+  getCognitoUserFromSub,
+  listGroupsForUser,
+  removeUserFromGroup,
+} from '~/lib/cognito.server'
+
+export async function loader({
+  params: { userId },
+  request,
+}: LoaderFunctionArgs) {
+  const currentUser = await getUser(request)
+  if (!currentUser?.groups.includes('gcn.nasa.gov/gcn-admin'))
+    throw new Response(null, { status: 403 })
+  if (!userId) throw new Response(null, { status: 400 })
+  const user = await getCognitoUserFromSub(userId)
+  const allGroups = await allGroupNames()
+  return { user, allGroups }
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const user = await getUser(request)
+  if (!user?.groups.includes('gcn.nasa.gov/gcn-admin'))
+    throw new Response(null, { status: 403 })
+  const data = await request.formData()
+  const { Username, ...selectedGroups } = Object.fromEntries(data)
+  if (!Username) throw new Response(null, { status: 400 })
```

Comment:
Is it necessary to include the username in the form data? Isn't it present in the route params?